src/util.c

```diff
@@ -730,3 +730,65 @@ ssize_t nonblock_read(int fd, void *buf, size_t count)
 
 	return len > 0 ? len : ret;
 }
+
+#define CONTROLLEN CMSG_LEN(sizeof(int))
+
+int hyper_send_fd(int fd, int fd_to_send) {
+	struct iovec	iov[1];
+	struct msghdr	msg;
+	char   buf[1];
+	struct cmsghdr	*cmptr = NULL;
+
+	iov[0].iov_base = buf;
+	iov[0].iov_len	= 1;
+	msg.msg_iov	= iov;
+	msg.msg_iovlen	= 1;
+	msg.msg_name	= NULL;
+	msg.msg_namelen	= 0;
+
+	if (fd_to_send < 0) {
+		return -1;
+	} else {
+		if ((cmptr = malloc(CONTROLLEN)) == NULL) {
```

Comment:
why do you malloc the memory, CONTROLLEN is constant, you can just define it in the stack